[Bug] 매니저 예약 조회 오류

### 🐛 버그 설명
무엇이 문제인지 명확하게 설명해주세요.
- 쿼리에서 manager의 예약을 조회할 때 reservationSchedule을 조인할 수 없다는 오류
---

### 🧪 재현 방법
어떻게 이 문제가 발생했는지 단계별로 작성해주세요.
1. 매니저의 예약 관리 페이지 접근
2. 조회된 예약이 없는 것을 확인
3. 브라우저 network에서 오류 코드 확인
4. 백엔드 로그에서 예약 조회시 조인 오류 확인

---

### 🎯 기대 동작
정상적으로 동작했어야 할 결과
- 특정 managerId로 조회한 예약 목록을 조회

---

### 📸 스크린샷 또는 로그(선택)
에러 메시지나 콘솔 로그, 화면 캡처 등
- 오류 로그
2025-07-04 11:33:25 [http-nio-8080-exec-10] ERROR c.k.g.c.h.GlobalExceptionHandler - Unexpected error occurred
org.springframework.dao.InvalidDataAccessApiUsageException: org.hibernate.query.PathException: Could not resolve join path 'reservationSchedule.reservation'

Caused by: org.hibernate.query.PathException: Could not resolve join path 'reservationSchedule.reservation'
	at org.hibernate.query.hql.internal.QualifiedJoinPathConsumer$ExpectingEntityJoinDelegate.consumeIdentifier(QualifiedJoinPathConsumer.java:356)

--- a/reservation/src/main/java/com/kernel/reservation/repository/CustomManagerReservationRepositoryImpl.java
+++ b/reservation/src/main/java/com/kernel/reservation/repository/CustomManagerReservationRepositoryImpl.java
@@ -2,6 +2,7 @@ package com.kernel.reservation.repository;
 
 import com.kernel.global.common.enums.UserRole;
 import com.kernel.global.domain.entity.QUser;
+import com.kernel.member.domain.entity.QUserInfo;
 import com.kernel.reservation.common.enums.MatchStatus;
 import com.kernel.reservation.domain.entity.*;
 import com.kernel.reservation.service.info.ManagerReservationDetailInfo;
@@ -33,6 +34,15 @@ import org.springframework.stereotype.Repository;
 public class CustomManagerReservationRepositoryImpl implements CustomManagerReservationRepository {
 
     private final JPQLQueryFactory jpaQueryFactory;
+    private final QUser user = QUser.user;
+    private final QUserInfo userInfo = QUserInfo.userInfo;
+    private final QReservation reservation = QReservation.reservation;
+    private final QReservationSchedule reservationSchedule = QReservationSchedule.reservationSchedule;
+    private final QServiceCheckLog serviceCheckLog = QServiceCheckLog.serviceCheckLog;
+    private final QReservationMatch reservationMatch = QReservationMatch.reservationMatch;
+    private final QServiceCategory serviceCategory = QServiceCategory.serviceCategory;
+    private final QExtraService extraService = QExtraService.extraService;
+
 
     /**
      * 매니저에게 할당된 예약 목록 조회 (검색 조건 및 페이징 처리)
@@ -45,64 +55,70 @@ public class CustomManagerReservationRepositoryImpl implements CustomManagerRese
     public Page<ManagerReservationSummaryInfo> searchManagerReservationsWithPaging(
             Long managerId, ManagerReservationSearchCondDTO searchCondDTO, Pageable pageable
     ) {
-        QReservation reservation = QReservation.reservation;
-        QReservationSchedule reservationSchedule = QReservationSchedule.reservationSchedule;
-        QServiceCheckLog serviceCheckLog = QServiceCheckLog.serviceCheckLog;
-        QReservationMatch reservationMatch = QReservationMatch.reservationMatch;
-
         // 전체 개수 조회
         long total = Optional.ofNullable(
             jpaQueryFactory
                 .select(reservation.count())
-                .from(reservation)
-                .leftJoin(reservationSchedule.reservation, reservation)
-                .leftJoin(serviceCheckLog.reservation, reservation)
-                .leftJoin(reservationMatch).on(reservationMatch.reservation.reservationId.eq(reservation.reservationId))
+                .from(reservation) // reservation을 루트로 설정
+                .leftJoin(reservationSchedule).on(reservationSchedule.reservation.eq(reservation))
+                .leftJoin(serviceCheckLog).on(serviceCheckLog.reservation.eq(reservation))
+                .leftJoin(reservationMatch).on(reservationMatch.reservation.eq(reservation))
+                .leftJoin(reservation.serviceCategory, serviceCategory)
+                .leftJoin(user).on(reservation.user.eq(user))
+                .leftJoin(userInfo).on(userInfo.user.eq(user))
                 .where(
-                    managerEq(managerId, reservation.reservationId.longValue(), reservationMatch), // 매니저 ID 일치
+                    reservationMatch.manager.userId.eq(managerId),
                     RequestDateGoe(searchCondDTO.getFromRequestDate()),         // 청소 예약 날짜 >= 시작일
                     RequestDateLoe(searchCondDTO.getToRequestDate()),           // 청소 예약 날짜 <= 종료일
-                    reservationStatus(searchCondDTO.getReservationStatus()),    // 예약 상태
-                    isCheckedIn(searchCondDTO.getIsCheckedIn()),                // 체크인 여부
-                    isCheckedOut(searchCondDTO.getIsCheckedOut()),              // 체크아웃 여부
-                    customerNameLike(searchCondDTO.getCustomerNameKeyword())         // 고객명 검색어 포함
+                    reservationStatus(searchCondDTO.getReservationStatus()),
+                    isCheckedIn(searchCondDTO.getIsCheckedIn()),
+                    isCheckedOut(searchCondDTO.getIsCheckedOut()),
+                    customerNameContains(searchCondDTO.getCustomerNameKeyword()),
+                    customerAddressContains(searchCondDTO.getCustomerAddressKeyword())
                 )
                 .fetchOne()
         ).orElse(0L);
 
         // 페이지 결과 조회
         List<ManagerReservationSummaryInfo> results = jpaQueryFactory
-                .select(Projections.fields(ManagerReservationSummaryInfo.class,
-                        reservation.reservationId,
-                        reservationSchedule.requestDate,
-                        reservationSchedule.startTime,
-                        reservationSchedule.turnaround,
-                        reservation.user.userId,
-                        reservation.serviceCategory.serviceName,
-                        reservation.status,
-                        serviceCheckLog.reservation,
-                        Expressions.booleanTemplate("CASE WHEN {0} IS NOT NULL THEN true ELSE false END", serviceCheckLog.inTime).as("isCheckedIn"),
-                        serviceCheckLog.inTime,
-                        Expressions.booleanTemplate("CASE WHEN {0} IS NOT NULL THEN true ELSE false END", serviceCheckLog.outTime).as("isCheckedOut"),
-                        serviceCheckLog.outTime
-                ))
-                .from(reservation)
-                .leftJoin(reservationSchedule.reservation, reservation)
-                .leftJoin(serviceCheckLog.reservation, reservation)
-                .leftJoin(reservationMatch).on(reservationMatch.reservation.reservationId.eq(reservation.reservationId))
-                .where(
-                        managerEq(managerId, reservation.reservationId.longValue(), reservationMatch),
-                        RequestDateGoe(searchCondDTO.getFromRequestDate()),
-                        RequestDateLoe(searchCondDTO.getToRequestDate()),
-                        reservationStatus(searchCondDTO.getReservationStatus()),
-                        isCheckedIn(searchCondDTO.getIsCheckedIn()),
-                        isCheckedOut(searchCondDTO.getIsCheckedOut()),
-                        customerNameLike(searchCondDTO.getCustomerNameKeyword())
-                )
-                .offset(pageable.getOffset())
-                .limit(pageable.getPageSize())
-                .orderBy(reservation.reservationId.desc())
-                .fetch();
+            .select(Projections.fields(ManagerReservationSummaryInfo.class,
+                reservation.reservationId,
+                reservationSchedule.requestDate,
+                reservationSchedule.startTime,
+                reservationSchedule.turnaround,
+                user.userId,
+                user.userName,
+                reservation.serviceCategory.serviceName,
+                reservation.status,
+                serviceCheckLog.reservation,
+                Expressions.booleanTemplate("CASE WHEN {0} IS NOT NULL THEN true ELSE false END", serviceCheckLog.inTime).as("isCheckedIn"),
+                serviceCheckLog.inTime,
+                Expressions.booleanTemplate("CASE WHEN {0} IS NOT NULL THEN true ELSE false END", serviceCheckLog.outTime).as("isCheckedOut"),
+                serviceCheckLog.outTime,
+                userInfo.roadAddress,
+                userInfo.detailAddress
+            ))
+            .from(reservation)
+            .leftJoin(reservationSchedule).on(reservationSchedule.reservation.eq(reservation))
+            .leftJoin(serviceCheckLog).on(serviceCheckLog.reservation.eq(reservation))
+            .leftJoin(reservationMatch).on(reservationMatch.reservation.eq(reservation))
+            .leftJoin(reservation.serviceCategory, serviceCategory)
+            .leftJoin(user).on(reservation.user.eq(user))
+            .leftJoin(userInfo).on(userInfo.user.eq(user))
+            .where(
+                reservationMatch.manager.userId.eq(managerId),
+                RequestDateGoe(searchCondDTO.getFromRequestDate()),
+                RequestDateLoe(searchCondDTO.getToRequestDate()),
+                reservationStatus(searchCondDTO.getReservationStatus()),
+                isCheckedIn(searchCondDTO.getIsCheckedIn()),
+                isCheckedOut(searchCondDTO.getIsCheckedOut()),
+                customerNameContains(searchCondDTO.getCustomerNameKeyword()),
+                customerAddressContains(searchCondDTO.getCustomerAddressKeyword())
+            )
+            .offset(pageable.getOffset())
+            .limit(pageable.getPageSize())
+            .orderBy(reservation.reservationId.desc())
+            .fetch();
 
         return new PageImpl<>(results, pageable, total);
     }
@@ -117,13 +133,6 @@ public class CustomManagerReservationRepositoryImpl implements CustomManagerRese
     public ManagerReservationDetailInfo findByManagerIdAndReservationId(
         Long managerId, Long reservationId
     ) {
-        QReservation reservation = QReservation.reservation;
-        QReservationSchedule reservationSchedule = QReservationSchedule.reservationSchedule;
-        QReservationMatch reservationMatch = QReservationMatch.reservationMatch;
-        QServiceCheckLog serviceCheckLog = QServiceCheckLog.serviceCheckLog;
-        QServiceCategory serviceCategory = QServiceCategory.serviceCategory;
-        QExtraService extraService = QExtraService.extraService;
-
 
         // 추가 서비스 조회
         Expression<String> extraServiceNameExpr = ExpressionUtils.as(
@@ -150,32 +159,35 @@ public class CustomManagerReservationRepositoryImpl implements CustomManagerRese
                 reservation.serviceCategory.serviceName,
                 reservation.status,
                 reservation.user.userId,
-                extraServiceNameExpr,
+                //extraServiceNameExpr,
                 reservation.memo,
                 serviceCheckLog.reservation,
                 serviceCheckLog.inTime,
                 serviceCheckLog.inFileId,
                 serviceCheckLog.outTime,
-                serviceCheckLog.outFileId
+                serviceCheckLog.outFileId,
+                user.userName,
+                userInfo.roadAddress,
+                userInfo.detailAddress
             ))
             .from(reservation)
-            .leftJoin(reservation.user).on(reservation.user.role.eq(UserRole.CUSTOMER))
-            .leftJoin(reservationSchedule.reservation, reservation)
-            .leftJoin(serviceCheckLog.reservation, reservation)
-            .leftJoin(reservationMatch).on(reservationMatch.reservation.reservationId.eq(reservation.reservationId))
+            .leftJoin(reservationSchedule).on(reservationSchedule.reservation.eq(reservation))
+            .leftJoin(serviceCheckLog).on(serviceCheckLog.reservation.eq(reservation))
+            .leftJoin(reservationMatch).on(reservationMatch.reservation.eq(reservation))
+            .leftJoin(reservation.serviceCategory, serviceCategory)
+            .leftJoin(user).on(reservation.user.eq(user))
+            .leftJoin(userInfo).on(userInfo.user.eq(user))
             .where(
                 reservation.reservationId.eq(reservationId),
-                managerEq(managerId, reservation.reservationId, reservationMatch) // 예약 매치 테이블에서 예약 ID가 reservationId와 일치하고 매칭상태가 MATCHED인 매니저 ID가 필요
+                reservationMatch.manager.userId.eq(managerId)
             )
             .fetchOne();
     }
 
-    // 매니저 ID 일치 ->  예약 매치 테이블에서 예약 ID가 reservationId와 일치하고 매칭상태가 MATCHED인 매니저 ID가 필요
-    private BooleanExpression managerEq(Long managerId, NumberExpression<Long> reservationId, QReservationMatch reservationMatch) {
-        return QUser.user.userId.eq(managerId)
-                .and(QUser.user.role.eq(UserRole.MANAGER))
-                .and(reservationMatch.reservation.reservationId.eq(reservationId))
-                .and(reservationMatch.status.eq(MatchStatus.MATCHED));
+    // 매니저 ID 일치 및 역할이 매니저인 경우
+    private BooleanExpression managerEq(Long managerId) {
+        return QUser.user.userId.eq(managerId);
+                //.and(QUser.user.role.eq(UserRole.MANAGER));
     }
 
     // 청소 예약 날짜 >= 시작일
@@ -194,6 +206,7 @@ public class CustomManagerReservationRepositoryImpl implements CustomManagerRese
 
     // 예약 상태
     private BooleanExpression reservationStatus(List<ReservationStatus> statuses) {
+        System.out.println("Reservation Statuses: " + statuses);
         return (statuses != null && !statuses.isEmpty())
             ? QReservation.reservation.status.in(statuses)
             : null;
@@ -224,10 +237,18 @@ public class CustomManagerReservationRepositoryImpl implements CustomManagerRese
     }*/
 
     // 고객명 검색어 포함 -> 검색된 이름 중에 role이 CUSTOMER인 경우만 조회 필요
-    private BooleanExpression customerNameLike(String keyword) {
+    private BooleanExpression customerNameContains(String keyword) {
         return (keyword != null && !keyword.isBlank())
-            ? QReservation.reservation.user.userName.eq(keyword)
+            ? QReservation.reservation.user.userName.contains(keyword)
                 .and(QReservation.reservation.user.role.eq(UserRole.CUSTOMER))
             : null;
+    }
+
+    // 고객 주소 검색어 포함
+    private BooleanExpression customerAddressContains(String keyword) {
+        return (keyword != null && !keyword.isBlank())
+                ? QUserInfo.userInfo.roadAddress.contains(keyword)
+                .or(QUserInfo.userInfo.detailAddress.contains(keyword))
+                : null;
     }
 }

--- a/reservation/src/main/java/com/kernel/reservation/service/info/ManagerReservationDetailInfo.java
+++ b/reservation/src/main/java/com/kernel/reservation/service/info/ManagerReservationDetailInfo.java
@@ -2,6 +2,7 @@ package com.kernel.reservation.service.info;
 
 import com.kernel.sharedDomain.common.enums.ReservationStatus;
 import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 
@@ -10,7 +11,8 @@ import java.time.LocalDate;
 import java.time.LocalTime;
 
 @Getter
-@NoArgsConstructor(access = AccessLevel.PROTECTED)
+@NoArgsConstructor
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
 public class ManagerReservationDetailInfo {
 
     private Long reservationId;
@@ -27,5 +29,8 @@ public class ManagerReservationDetailInfo {
     private Long inFileId;
     private Timestamp outTime;
     private Long outFileId;
+    private String roadAddress;
+    private String detailAddress;
+    private String userName;
 
 }

--- a/reservation/src/main/java/com/kernel/reservation/service/info/ManagerReservationSummaryInfo.java
+++ b/reservation/src/main/java/com/kernel/reservation/service/info/ManagerReservationSummaryInfo.java
@@ -2,6 +2,7 @@ package com.kernel.reservation.service.info;
 
 import com.kernel.sharedDomain.common.enums.ReservationStatus;
 import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 
@@ -10,14 +11,16 @@ import java.time.LocalDate;
 import java.time.LocalTime;
 
 @Getter
-@NoArgsConstructor(access = AccessLevel.PROTECTED)
+@NoArgsConstructor
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
 public class ManagerReservationSummaryInfo {
 
     private Long reservationId;
     private LocalDate requestDate;
     private LocalTime startTime;
     private Integer turnaround;
-    private Long customerId;
+    private Long userId;
+    private String userName;
     private String serviceName;
     private ReservationStatus status;
     private Long serviceCheckId;
@@ -25,5 +28,7 @@ public class ManagerReservationSummaryInfo {
     private Timestamp inTime;
     private Boolean isCheckedOut;
     private Timestamp outTime;
+    private String roadAddress;
+    private String detailAddress;
 
 }

--- a/reservation/src/main/java/com/kernel/reservation/service/request/ManagerReservationSearchCondDTO.java
+++ b/reservation/src/main/java/com/kernel/reservation/service/request/ManagerReservationSearchCondDTO.java
@@ -24,7 +24,7 @@ public class ManagerReservationSearchCondDTO {
     @Schema(description = "예약 날짜 종료일", example = "2023-01-31", required = true)
     private LocalDate toRequestDate;
 
-    @Schema(description = "예약 상태 목록", example = "[\"CONFIRMED\", \"CANCELLED\"]", required = false)
+    @Schema(description = "예약 상태 목록", example = "[\"REQUESTED\", \"CONFIRMED\", \"COMPLETED\", \"CANCELED\"]", required = false)
     private List<ReservationStatus> reservationStatus;
 
     @Schema(description = "체크인 여부", example = "true", required = false)
@@ -38,4 +38,7 @@ public class ManagerReservationSearchCondDTO {
 
     @Schema(description = "고객명 키워드", example = "홍길동", required = false)
     private String customerNameKeyword;
+
+    @Schema(description = "고객 주소 키워드", example = "서울특별시 강남구", required = false)
+    private String customerAddressKeyword;
 }

--- a/reservation/src/main/java/com/kernel/reservation/service/response/ManagerReservationRspDTO.java
+++ b/reservation/src/main/java/com/kernel/reservation/service/response/ManagerReservationRspDTO.java
@@ -58,6 +58,15 @@ public class ManagerReservationRspDTO {
     @Schema(description = "체크아웃 첨부파일 ID", example = "102", required = false)
     private Long outFileId;
 
+    @Schema(description = "고객 도로명 주소", example = "서울특별시 강남구 테헤란로 123", required = true)
+    private String roadAddress;
+
+    @Schema(description = "고객 상세 주소", example = "101동 202호", required = true)
+    private String detailAddress;
+
+    @Schema(description = "고객 이름", example = "홍길동", required = true)
+    private String userName;
+
     @Schema(description = "ManagerReservationDetailInfo에서 필요한 필드만 포함하여 DTO로 변환")
     public static ManagerReservationRspDTO fromInfo(ManagerReservationDetailInfo info) {
         return ManagerReservationRspDTO.builder()
@@ -75,6 +84,9 @@ public class ManagerReservationRspDTO {
                 .inFileId(info.getInFileId())
                 .outTime(info.getOutTime() != null ? info.getOutTime().toLocalDateTime() : null)
                 .outFileId(info.getOutFileId())
+                .roadAddress(info.getRoadAddress())
+                .detailAddress(info.getDetailAddress())
+                .userName(info.getUserName())
                 .build();
     }
 

--- a/reservation/src/main/java/com/kernel/reservation/service/response/ManagerReservationSummaryRspDTO.java
+++ b/reservation/src/main/java/com/kernel/reservation/service/response/ManagerReservationSummaryRspDTO.java
@@ -5,6 +5,7 @@ import java.time.LocalTime;
 
 import com.kernel.reservation.service.info.ManagerReservationSummaryInfo;
 import com.kernel.sharedDomain.common.enums.ReservationStatus;
+import com.kernel.sharedDomain.domain.entity.ServiceCategory;
 import io.swagger.v3.oas.annotations.media.Schema;
 import com.kernel.sharedDomain.common.enums.ReservationStatus;
 import io.swagger.v3.oas.annotations.media.Schema;
@@ -30,6 +31,9 @@ public class ManagerReservationSummaryRspDTO {
     @Schema(description = "청소 요청 날짜", example = "2023-01-01", required = true)
     private LocalDate requestDate;
 
+    @Schema(description = "서비스 이름", example = "일반 청소", required = true)
+    private String serviceName;
+
     @Schema(description = "예약 시간", example = "14:00", required = true)
     private LocalTime startTime;
 
@@ -37,7 +41,16 @@ public class ManagerReservationSummaryRspDTO {
     private Integer turnaround;
 
     @Schema(description = "고객 ID", example = "456", required = true)
-    private Long customerId;
+    private Long userId;
+
+    @Schema(description = "고객 이름", example = "홍길동", required = true)
+    private String userName;
+
+    @Schema(description = "고객 도로명 주소", example = "서울특별시 강남구 테헤란로 123", required = true)
+    private String customerRoadAddress;
+
+    @Schema(description = "고객 상세 주소", example = "101동 202호", required = true)
+    private String customerDetailAddress;
 
     @Schema(description = "예약 상태", example = "CONFIRMED", required = true)
     private ReservationStatus status;
@@ -62,9 +75,13 @@ public class ManagerReservationSummaryRspDTO {
         return info.map(reservation -> ManagerReservationSummaryRspDTO.builder()
                 .reservationId(reservation.getReservationId())
                 .requestDate(reservation.getRequestDate())
+                .serviceName(reservation.getServiceName())
                 .startTime(reservation.getStartTime())
                 .turnaround(reservation.getTurnaround())
-                .customerId(reservation.getCustomerId())
+                .userId(reservation.getUserId())
+                .userName(reservation.getUserName())
+                .customerRoadAddress(reservation.getRoadAddress())
+                .customerDetailAddress(reservation.getDetailAddress())
                 .status(reservation.getStatus())
                 .reservationCheckId(reservation.getServiceCheckId())
                 .isCheckedIn(reservation.getIsCheckedIn())

